src/Components/test/testassets/BasicTestApp/ModifyHeadComponent.razor

```diff
@@ -0,0 +1,119 @@
+﻿@using Microsoft.AspNetCore.Components.Web.Extensions
+
+<p>
+    Multiple title elements:<br />
+
+    @for (int i = 0; i < 3; i++)
+    {
+        var titleId = i;
+
+        <button id="button-title-@i" @onclick="() => SetSelectedTitle(titleId)">
+            Title @titleId
+        </button>
+
+        if (selectedTitle == titleId)
+        {
+            <Title Value="@($"Title {titleId}")" />
+        }
+    }
+</p>
+
+<p>
+    Multiple meta elements:<br />
+
+    @for (int i = 0; i < metas.Length; i++)
+    {
+        var metaId = i;
+
+        <button id="button-meta-@i" @onclick="() => Toggle(ref metas[metaId])">
```

Comment:
It's not common (nor recommended I would say) to use ref within a handler. I get what you are trying to do, but I would generally recommend against it since you are saving writing an extra method at the cost of making it harder for people to understand what's going on.

An alternative that would also work without requiring ref, is to simply pass the array (metas or links) as an additional parameter.